packages/bot-web-ui/src/pages/bot-builder/quick-strategy/form-wrappers/upgraded-qs-v2/desktop-form-wrapper.tsx

```diff
@@ -0,0 +1,204 @@
+};
+
+const QuickSelectionPanel = ({
+    selected_trade_type,
+    selected_startegy_label,
+    children,
+}: Pick<TDesktopFormWrapper, 'selected_trade_type' | 'children'> & { selected_startegy_label: string }) => (
+    <>
+        <div className='qs__selected-options'>
+            <div className='qs__selected-options__item'>
+                <Text size='xs'>{localize('Trade type')}</Text>
+                <Text size='xs' weight='bold'>
+                    {selected_trade_type}
+                </Text>
+            </div>
+            <div className='qs__selected-options__item'>
+                <Text size='xs'>{localize('Strategy')}</Text>
+                <Text size='xs' weight='bold'>
+                    {selected_startegy_label}
+                </Text>
+            </div>
+        </div>
+        <StrategyTabContent formfields={children} active_tab={'TRADE_PARAMETERS'} />
+    </>
+);
+
+const FormWrapper = observer(
+    ({
+        children,
+        current_step,
+        setCurrentStep,
+        onClickClose,
+        selected_trade_type,
+        setSelectedTradeType,
+    }: TDesktopFormWrapper) => {
+        const scroll_ref = React.useRef<HTMLDivElement & SVGSVGElement>(null);
+        const { submitForm, isValid, setFieldValue, validateForm, values } = useFormikContext<TFormValues>();
+        const { quick_strategy } = useDBotStore();
+        const { selected_strategy, onSubmit, is_stop_bot_dialog_open } = quick_strategy;
+        const { handleSubmit } = useQsSubmitHandler();
+
+        const selected_startegy_label = STRATEGIES[selected_strategy as keyof typeof STRATEGIES].label;
+        const is_selected_strategy_step = current_step === QsSteps.StrategySelect;
+
+        React.useEffect(() => {
+            if (isValid && current_step === QsSteps.StrategyVerified) {
+                setCurrentStep(QsSteps.StrategyCompleted);
+            }
+            if (!isValid && current_step === QsSteps.StrategyCompleted) {
+                setCurrentStep(QsSteps.StrategyVerified);
+            }
+        }, [isValid, current_step]);
+
+        React.useEffect(() => {
+            validateForm();
+        }, [selected_strategy, validateForm]);
+
+        const onEdit = async () => {
+            await setFieldValue('action', 'EDIT');
+            validateForm();
+            submitForm().then((form_data: TFormData | void) => {
+                if (isValid && form_data) {
+                    rudderStackSendQsEditStrategyEvent({
+                        form_values: values,
+                        selected_strategy,
+                    });
+                    onSubmit(form_data); // true to load and run the bot
+                }
+            });
+        };
+
+        const onRun = () => {
+            handleSubmit();
+        };
+
+        const onBack = () => {
+            setCurrentStep(QsSteps.StrategySelect);
+        };
+
+        const renderContent = () => {
```

Comment:
can we use useCallback here for better performance ?

Reply:
I've added it, but since the component re-renders frequently you might not see significant benefits of usage useCallback in this particular case.